File: loopr-ios/Models/Order/OrderState.swift
//
//  OrderStatus.swift
//  loopr-ios
//
//  Created by xiaoruby on 2/5/18.
//  Copyright © 2018 Loopring. All rights reserved.
//

import UIKit
import BigInt

class OrderState {

    var status: OrderStatus = .pending

    // tokens可用于订单数量，暂时无用
    var actualAmountS: String = ""

    var actualAmountB: String = ""

    var actualAmountFee: String = ""

    // 订单未成交数量，计算fill
    var outstandingAmountS: String = ""

    var outstandingAmountSell: Double

    var outstandingAmountB: String = ""

    var outstandingAmountBuy: Double

    var outstandingAmountFee: String = ""

    var outstandingAmountF: Double

    init() {

    }

    init(json: JSON) {
        self.status = OrderStatus(rawValue: json["status"].stringValue) ?? .unknown
        self.actualAmountS = json["actualAmountS"].stringValue
        self.actualAmountB = json["actualAmountB"].stringValue
        self.actualAmountFee = json["actualAmountFee"].stringValue
        self.outstandingAmountS = json["outstandingAmountS"].stringValue
        self.outstandingAmountB = json["outstandingAmountB"].stringValue
        self.outstandingAmountFee = json["outstandingAmountFee"].stringValue
    }
}

File: walletsdk/src/main/java/leaf/prod/walletsdk/model/order/OrderState.java
/**
 * Created with IntelliJ IDEA.
 * User: kenshin wangchen@loopring.org
 * Time: 2018-11-16 11:34 AM
 * Cooperation: loopring.org 路印协议基金会
 */
package leaf.prod.walletsdk.model.order;

import java.io.Serializable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class OrderState implements Serializable {

    @SerializedName(value = "status")
    private OrderStatus status;

    // tokens可用于订单数量，暂时无用
    @SerializedName(value = "actualAmountS")
    private String actualAmountS;

    @SerializedName(value = "actualAmountB")
    private String actualAmountB;

    @SerializedName(value = "actualAmountFee")
    private String actualAmountFee;

    // 订单未成交数量，计算fill
    @SerializedName(value = "outstandingAmountS")
    private String outstandingAmountS;

    @Expose(serialize = false, deserialize = false)
    private Double outstandingAmountSell;

    @SerializedName(value = "outstandingAmountB")
    private String outstandingAmountB;

    @Expose(serialize = false, deserialize = false)
    private Double outstandingAmountBuy;

    @SerializedName(value = "outstandingAmountFee")
    private String outstandingAmountFee;

    @Expose(serialize = false, deserialize = false)
    private Double outstandingAmountF;
}
